src/components/Personalization/createApplyPropositions.js

```diff
@@ -0,0 +1,38 @@
+/*
+Copyright 2022 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import composePersonalizationResultingObject from "./utils/composePersonalizationResultingObject";
+import isNonEmptyArray from "../../utils/isNonEmptyArray";
+import isNonEmptyString from "../../utils/isNonEmptyString";
+
+export default ({ viewCache, executeDecisions, showContainers }) => {
+  const applyPropositions = ({ propositions }) => {
+    return executeDecisions(propositions).then(() => {
+      showContainers();
+      return composePersonalizationResultingObject(propositions, true);
+    });
+  };
+
+  return ({ propositions, viewName }) => {
```

Comment:
Consider when propositions and viewName are both specified. As written here, only the propositions would be rendered. Is this the desired behavior? If so we should log a warning to explain that only one should be specified and that only the propositions in the array will be applied.

Reply:
I don't think there is a valid use case where a user should send both viewName and propositions.  SPAs should only specify viewName and get the propositions from the view cache.  Non-SPAs should only specify propositions that should get rendered.  We'll definitely need to document this.